internal/reconcilers/controller/tasks/integrations/deployment.go

```diff
@@ -0,0 +1,141 @@
+package integrations
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pier-oliviert/konditionner/pkg/konditions"
+	phonebook "github.com/pier-oliviert/phonebook/api/v1alpha1"
+	"github.com/pier-oliviert/phonebook/api/v1alpha1/integrations"
+	"github.com/pier-oliviert/phonebook/api/v1alpha1/references"
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/env"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type deployment struct {
+	ctx         context.Context
+	integration *phonebook.DNSIntegration
+	client.Client
+}
+
+func DeploymentTask(ctx context.Context, c client.Client, integration *phonebook.DNSIntegration) konditions.Task {
+	t := deployment{
+		ctx:         ctx,
+		integration: integration,
+		Client:      c,
+	}
+
+	return t.Run
+}
+
+func (t deployment) Run(condition konditions.Condition) (konditions.Condition, error) {
+	d := t.deployment()
+	if err := t.Create(t.ctx, d); err != nil {
+		return condition, err
+	}
+
+	t.integration.Status.Deployment = references.NewReference(d)
+	condition.Status = konditions.ConditionCreated
+	condition.Reason = fmt.Sprintf("Deployment Created: %s", d.Name)
+
+	return condition, nil
+}
+
+func (t deployment) deployment() *apps.Deployment {
+	img := ""
+	if t.integration.Spec.Provider.Image != nil {
+		img = *t.integration.Spec.Provider.Image
+	} else {
+		img = fmt.Sprintf("ghcr.io/pier-oliviert/providers-%s:latest", t.integration.Spec.Provider.Name)
```

Comment:
I do wonder if `latest` is a bit risky, in theory if we released a newer image with breaking changes, and someone is running an older version of phonebook that restarts or re-pulls the image, it could pull the wrong image down

Perhaps we can pin the image versions somehow?

Reply:
I agree, when I did it I couldn't settle on a tag. And now that you raise this issue, I think I was trying to be too clever with this. What do you think of having a hash in the go binary that list providers with their latest stable version of the image. Something like this:

```go
var ProviderList = map[string]string{
  aws: "ghcr.io/pier-oliviert/providers-aws:0.0.1",
  azure: "ghcr.io/pier-oliviert/providers-azure:0.1",
  // etc.
}
```

What do you think? It's also always possible for the user to specify the image themselves too, if they ever needed to.

Reply:
I like the approach, we can make sure compatability out of the box works and users can still go to a custom image version / older / whatever if they really need to